tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionSettingsTests.cs

```diff
@@ -168,25 +184,28 @@ public void With_applicationName_should_return_expected_result()
             var result = subject.With(applicationName: newApplicationName);
 
             result.ApplicationName.Should().Be(newApplicationName);
-            result.Authenticators.Should().Equal(subject.Authenticators);
+            AssertActualAuthenticatorsFactoryResult(result.AuthenticatorFactories, subject.AuthenticatorFactories);
             subject.Compressors.Should().Equal(__defaults.Compressors);
             result.MaxIdleTime.Should().Be(subject.MaxIdleTime);
             result.MaxLifeTime.Should().Be(subject.MaxLifeTime);
         }
 
         [Fact]
-        public void With_authenticators_should_return_expected_result()
+        public void With_authenticatorsFactory_should_return_expected_result()
         {
 #pragma warning disable 618
-            var oldAuthenticators = new[] { new MongoDBCRAuthenticator(new UsernamePasswordCredential("source", "username1", "password1")) };
-            var newAuthenticators = new[] { new MongoDBCRAuthenticator(new UsernamePasswordCredential("source", "username2", "password2")) };
+            var oldAuthenticator = new MongoDBCRAuthenticator(new UsernamePasswordCredential("source", "username1", "password1"));
+            var newAuthenticator = new MongoDBCRAuthenticator(new UsernamePasswordCredential("source", "username2", "password2"));
 #pragma warning restore 618
-            var subject = new ConnectionSettings(authenticators: oldAuthenticators);
+            var oldAuthenticatorFactories = new[] { new AuthenticatorFactory(() => oldAuthenticator) };
+            var newAuthenticatorFactories = new[] { new AuthenticatorFactory(() => newAuthenticator) }; ;
 
-            var result = subject.With(authenticators: newAuthenticators);
+            var subject = new ConnectionSettings(authenticatorFactories: oldAuthenticatorFactories);
+
+            var result = subject.With(authenticatorFactories: newAuthenticatorFactories);
 
             result.ApplicationName.Should().Be(subject.ApplicationName);
-            result.Authenticators.Should().Equal(newAuthenticators);
+            AssertActualAuthenticatorsFactoryResult(result.AuthenticatorFactories, subject.AuthenticatorFactories);
```

Comment:
Replace with:

            result.AuthenticatorFactories.Should().Equal(newAuthenticatorFactories);